src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -98,35 +88,43 @@ abstract class ManagedStrategy(
     // used on resumption, and the trace point before and after the suspension
     // correspond to the same method call in the trace.
     private val suspendedFunctionsStack = Array(nThreads) { mutableListOf<Int>() }
+    // Helps to ignore potential switch point in local objects (see LocalObjectManager) to avoid
+    // useless interleavings analysis
+    private var localObjectManager = LocalObjectManager()
+    // Last read trace point, occurred in the current thread.
+    // We store it as we initialize read value after the point is created so we have to store
+    // the trace point somewhere to obtain it later
+    private var lastReadTracePoint = ThreadLocal<ReadTracePoint?>()
```

Comment:
Why thread-local instead of an array? We always use arrays in this code.

Reply:
Please also indicate why exactly you need to store it, because now it's unclear.

Reply:
There was no reason to store it this way, so now it's stored in Array, like everywhere.